### PR TITLE
Mobile Experience

### PR DESCRIPTION
**Pull request by SunriseFox**
Known issues:

+ [ ] `Add it for me` button overflows its container
+ [ ] `Welcome-Step` page overflows its container
+ [x] (iOS) the immersive setup popup is not draggable @Jack-Works (might be fixed by https://github.com/DimensionDev/webextension-polyfill/commit/7324671bfde3c02016dc22193480b4bf8e33466b)
+ [x] (Android) options page won't run `browser.tabs.update`

Known but more information is needed to identify what and why: 

+ [x] a chance that when clicking `Connect facebook.com`, switched to the Facebook page but nothing else happened.
+ [x] a chance that when clicking `Connect facebook.com`, the Facebook page is nearly not operatable at all, with a very high CPU usage. (Might be fixed by https://github.com/DimensionDev/Maskbook/pull/860)
+ [x] a chance when visiting others' Facebook pages then do connect, the auto-detected name is not ours but theirs.
+ [x] when immersive setup procedure is finished, click `Maskbook Settings`, the Dashboard still shows a state that no profiles are connected to that persona.

Test:

+ [x] (Android) a new options page is opened whenever clicking `Maskbook Settings`

### CODE REVIEW

**Comment by Tedko** [comment]
IMG_6353
IMG_6352


Some overflow for texts in dashboard (eg. choose eth network/ home-import)

**Comment by Tedko** [comment]
IMG_5188 PNG

Overflow on iPhone X (testflight 1.10.9(23) )

File: src/extension/content-script/tasks.ts [hunk 1]
```diff
@@ -67,6 +67,57 @@ export default function tasks(...args: Parameters<typeof realTasks>) {
...
+const uriCanDoTask = (tabUri?: string, targetUri?: any) => {
+    if (tabUri?.startsWith(targetUri)) return true
+    return false
+}
+
+/**
+ * ! For mobile standalone app
+ * This function will open/switch tabs (and start tasks)
+ * It tries to check if a tab being able to do the target task already opened before opening it
+ * For Chrome, `browser.tabs.query` won't work for chrome-extension: schema, so this function is not useful
```
Comment (Jack-Works) [inline src/extension/content-script/tasks.ts]: Does it work on the Firefox browser?

File: src/extension/content-script/tasks.ts [hunk 2]
```diff
@@ -67,6 +67,57 @@ export default function tasks(...args: Parameters<typeof realTasks>) {
...
+ */
+
+export function exclusiveTasks(...args: Parameters<typeof realTasks>) {
```
Comment (guanbinrui) [inline src/extension/content-script/tasks.ts]: Why not merge this method into `tasks`? No need for the caller knowing about the current build target.

File: src/extension/options-page/DashboardComponents/ProfileBox.tsx [hunk 3]
```diff
@@ -72,15 +71,22 @@ export default function ProfileBox({ persona, border }: Props) {
...
-        return {
-            network: i.networkIdentifier,
-            connected: !!profile,
-            userId: profile?.[0].userId,
-            identifier: profile?.[0],
-        }
-    })
+    const providers = [...definedSocialNetworkWorkers]
+        .filter(i => {
+            if (webpackEnv.genericTarget === 'facebookApp') {
```
Comment (guanbinrui) [inline src/extension/options-page/DashboardComponents/ProfileBox.tsx]: Can be simplied as `webpackEnv.genericTarget !== 'facebookApp'`

File: src/components/shared-settings/useSettingsUI.tsx [hunk 4]
```diff
@@ -56,6 +60,11 @@ export function SettingsUI<T>(props: SettingsUIProps<T>) {
             )
     }
 }
+
+const useStyles2 = makeStyles({
```
Comment (guanbinrui) [inline src/components/shared-settings/useSettingsUI.tsx]: Merge this style into the previous one? Or another name?

File: src/extension/content-script/tasks.ts (hunk 1 above)
Comment (SunriseFox) [reply src/extension/content-script/tasks.ts]: work

File: src/extension/options-page/DashboardComponents/ProfileBox.tsx (hunk 3 above)
Comment (SunriseFox) [reply src/extension/options-page/DashboardComponents/ProfileBox.tsx]: maybe current is more clear

File: src/extension/content-script/tasks.ts (hunk 2 above)
Comment (SunriseFox) [reply src/extension/content-script/tasks.ts]: the word `exclusive` means its target URL is exclusive on mobile... for example, we should not open another tab if there is already one can do the task. we don't utilize this feature on the extension version now.

merge this into `tasks` is ok too but it seems not offering any benefit.

File: src/components/shared-settings/useSettingsUI.tsx (hunk 4 above)
Comment (SunriseFox) [reply src/components/shared-settings/useSettingsUI.tsx]: renamed

File: src/extension/content-script/tasks.ts (hunk 1 above)
Comment (Jack-Works) [reply src/extension/content-script/tasks.ts]: 🤔

**Pull request merged**